### ISSUE TITLE
Initialisation of batchHandlers with empty HashMap skips default camunda initialisation

### ISSUE THREAD

**Issue by ronalders** [camunda/camunda-bpm-spring-boot-starter#196]
In DefaultFailedJobConfiguration initialises the batchHandlers with an empty HashMap:

```
if (configuration.getBatchHandlers() == null) {
      configuration.setBatchHandlers(new HashMap<>());
}
```

Because of this unnecessary initialisation the default batchHandlers initialisation of camunda is skipped. This causes missing batchJobHandlers (like Processmigration etc). 

```
protected void initBatchHandlers() {
    if (batchHandlers == null) {
      batchHandlers = new HashMap<String, BatchJobHandler<?>>();

      MigrationBatchJobHandler migrationHandler = new MigrationBatchJobHandler();
      batchHandlers.put(migrationHandler.getType(), migrationHandler);

      DeleteProcessInstancesJobHandler deleteProcessJobHandler = new DeleteProcessInstancesJobHandler();
      batchHandlers.put(deleteProcessJobHandler.getType(), deleteProcessJobHandler);

      DeleteHistoricProcessInstancesJobHandler deleteHistoricProcessInstancesJobHandler = new DeleteHistoricProcessInstancesJobHandler();
      batchHandlers.put(deleteHistoricProcessInstancesJobHandler.getType(), deleteHistoricProcessInstancesJobHandler);

      SetJobRetriesJobHandler setJobRetriesJobHandler = new SetJobRetriesJobHandler();
      batchHandlers.put(setJobRetriesJobHandler.getType(), setJobRetriesJobHandler);
    }

    if (customBatchJobHandlers != null) {
      for (BatchJobHandler<?> customBatchJobHandler : customBatchJobHandlers) {
        batchHandlers.put(customBatchJobHandler.getType(), customBatchJobHandler);
      }
    }
  }
```

And makes async processmigration impossible at the moment. The solution is simple, only  initialise the postBPMNParseListener because that is the list we are adding a listener to. Should I create a PR?

**Comment by jangalinski** [comment]
please go ahead ... we must have missed that ...

**Comment by jangalinski** [comment]
see https://forum.camunda.org/t/all-users-have-admin-access-with-spring-boot/2878/3

**Comment by jangalinski** [comment]
fixed with 63efa4e742dfe9ea5fe8c54ff4e69c0dff7a994e
